[Tablet support M2] Full flow

AnirudhBhat opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #10568 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR introduces a `nav_host_fragment` within the `OrderListFragment`, facilitating a two-pane layout for tablets. On phones, this layout adjustment will have no impact. The `nav_host_fragment` is designated to host the Order detail screen.

**Note:** This PR solely focuses on layout adjustments and does not encompass any business logic for selecting orders or automatically opening the first order by default.

### Testing instructions
<!-- Step-by-step testing instructions. When necessary, break out individual scenarios that need testing, and consider including a checklist for the reviewer to go through. -->
1. Run the app on physical tablet or in a tablet emulator
2. Click on the `Orders` tab
3. Ensure you see a 2 pane layout with order list and order detail
4. Go into split mode and resize the window and ensure the layout moves between 2 pane and single pane layout according to the window size
5. Run the same app on phone and ensure the behaviour is same as before and there's no change

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
<img width="859" alt="Screenshot 2024-01-23 at 9 01 26 AM" src="https://github.com/woocommerce/woocommerce-android/assets/1331230/e387f094-e772-4837-9883-83ef6a1dad6e">


- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

dangermattic commented:
<!--
  0 Errors
  3 Warnings: This PR is larger than 300 lin...
  1 Message: <p>This PR contains changes to...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>SelectedOrderTrackerViewModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/195">17.5</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

AnirudhBhat committed:
Add fragment specific toolbar for ShippingCustomsFragment.kt

AnirudhBhat committed:
Add fragment specific toolbar for ShippingCarrierRatesFragment.kt

AnirudhBhat committed:
Add fragment specific toolbar for CreateShippingLabelFragment.kt

AnirudhBhat committed:
Add fragment specific toolbar for EditShippingLabelAddressFragment.kt

AnirudhBhat committed:
Add fragment specific toolbar for EditShippingLabelPackagesFragment.kt

AnirudhBhat committed:
Add fragment specific toolbar for EditShippingLabelPackagesFragment.kt

AnirudhBhat committed:
Add fragment specific toolbar for EditShippingLabelPaymentFragment.kt

AnirudhBhat committed:
Fix crash on config change when searching in order list

AnirudhBhat committed:
Fix navigating to the web view from the order detail screen in a 2 pane layout

AnirudhBhat committed:
Fix navigating to the web view from the order detail screen in a 2 pane layout

AnirudhBhat committed:
Fix navigating to the info screen fragment

AnirudhBhat committed:
Add fragment specific toolbar for SelectPaymentMethodFragment.kt

AnirudhBhat committed:
Add fragment specific toolbar for PaymentsHubFragment.kt

AnirudhBhat committed:
Add fragment specific toolbar for CardReaderOnboardingFragment.kt

AnirudhBhat committed:
Add refreshOrders live data in SelectedOrderTrackerViewModel.kt which updates its observers to refresh the orders so that the latest order data is reflected on the UI

AnirudhBhat committed:
Observe the refresh orders live data in selected order tracker view model and refresh the orders when the order status changes

AnirudhBhat committed:
Add logic to open first order if the order list is freshly opened, open created order if order list is displayed as a result of new order creation

AnirudhBhat committed:
Add an optional argument orderId to OrderListFragment - it is required to open a newly created order from order creation screen.

AnirudhBhat committed:
Fix crash on CustomOrderFieldsFragment.kt and add toolbar

samiuelson commented on WooCommerce/src/main/res/layout/fragment_order_list.xml:
```diff
@@ -1,81 +1,110 @@
+
+    </com.woocommerce.android.widgets.ScrollChildSwipeRefreshLayout>
+
+    <LinearLayout
```
❓ Why do we need to wrap <fragment> with LinearLayout?

AnirudhBhat replied:
Good point. Removed unnecessary LinearLayout here: https://github.com/woocommerce/woocommerce-android/pull/10591/commits/ee6de645c029fedff0c59d1f002f81cfa4635e1d

samiuelson reviewed (commented):
Looks good overall. I wonder if we could simplify the XML layout slightly. Let me know what you think.

samiuelson reviewed (approved):
LGTM

AnirudhBhat committed:
Add toolbars to refund screens

AnirudhBhat committed:
Add toolbar elevation

AnirudhBhat committed:
Remove unused imports

AnirudhBhat committed:
Remove unused imports

AnirudhBhat committed:
Remove unused imports

AnirudhBhat committed:
Revert "Add fragment specific toolbar for EditShippingLabelPackagesFragment.kt"

This reverts commit 1210e19a805c0b908c1ecb7a8d00605f61a1ec66.

AnirudhBhat committed:
Remove unused imports

AnirudhBhat committed:
Remove unused imports

AnirudhBhat committed:
Fix detekt errors

AnirudhBhat committed:
Fix detekt error

AnirudhBhat committed:
add toolbar title in Shipping Address Fragment

AnirudhBhat committed:
Add a new key that is used to track payment flow start from the order creation screen

AnirudhBhat committed:
Make code changes so that when we try to collect a payment on a new order in the order creation screen, the screen is navigated to the select payment method on both tablet and phone

AnirudhBhat committed:
Fix detekt errors

AnirudhBhat committed:
Remove shared toolbar from BaseOrderEditingFragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-order-list-detail' into issue/toolbar-customer-order-note-editing

AnirudhBhat committed:
Merge branch 'issue/toolbar-customer-order-note-editing' into issue/toolbar-base-address-editing-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-base-address-editing-fragment' into issue/toolbar-add-order-note-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-add-order-note-fragment' into issue/toolbar-print-shipping-label-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-print-shipping-label-fragment' into issue/toolbar-label-format-options-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-label-format-options-fragment' into issue/toolbar-print-shipping-label-info-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-print-shipping-label-info-fragment' into issue/toolbar-print-shipping-label-customs-form-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-print-shipping-label-customs-form-fragment' into issue/toolbar-shipping-label-refund-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-shipping-label-refund-fragment' into issue/toolbar-shipping-package-selector

AnirudhBhat committed:
Merge branch 'issue/toolbar-shipping-package-selector' into issue/toolbar-create-package-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-create-package-fragment' into issue/toolbar-create-custom-package

AnirudhBhat committed:
Merge branch 'issue/toolbar-create-custom-package' into issue/toolbar-address-suggestion-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-address-suggestion-fragment' into issue/customs-fragment-shipping

AnirudhBhat committed:
Merge branch 'issue/customs-fragment-shipping' into issue/toolbar-carrier-rates-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-carrier-rates-fragment' into issue/toolbar-create-shipping-label

AnirudhBhat committed:
Merge branch 'issue/toolbar-create-shipping-label' into issue/toolbar-edit-shipping-label

AnirudhBhat committed:
Merge branch 'issue/toolbar-edit-shipping-label' into issue/toolbar-edit-shipping-label-package

AnirudhBhat committed:
Merge branch 'issue/toolbar-edit-shipping-label-package' into issue/toolbar-edit-shipping-payment

AnirudhBhat committed:
Merge branch 'issue/toolbar-edit-shipping-payment' into issue/toolbar-select-payment-method

AnirudhBhat committed:
Merge branch 'issue/toolbar-select-payment-method' into issue/toolbar-payments-hub-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-payments-hub-fragment' into issue/toolbar-card-reader-onboarding-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-card-reader-onboarding-fragment' into issue/refresh-orders-list

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListFragment.kt

AnirudhBhat committed:
Merge branch 'issue/refresh-orders-list' into issue/fix-custom-fields-crash

AnirudhBhat committed:
Merge branch 'issue/fix-custom-fields-crash' into issue/toolbar-refund-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-refund-fragment' into issue/toolbar-elevation

# Conflicts:
#	WooCommerce/src/main/res/layout/fragment_add_order_note.xml
#	WooCommerce/src/main/res/layout/fragment_card_reader_onboarding.xml
#	WooCommerce/src/main/res/layout/fragment_edit_shipping_label_payment.xml
#	WooCommerce/src/main/res/layout/fragment_label_format_options.xml
#	WooCommerce/src/main/res/layout/fragment_order_create_edit_customer_note.xml
#	WooCommerce/src/main/res/layout/fragment_order_detail.xml
#	WooCommerce/src/main/res/layout/fragment_payments_hub.xml
#	WooCommerce/src/main/res/layout/fragment_select_payment_method.xml
#	WooCommerce/src/main/res/layout/fragment_shipping_carrier_rates.xml
#	WooCommerce/src/main/res/layout/fragment_shipping_customs.xml
#	WooCommerce/src/main/res/layout/fragment_shipping_label_address_suggestion.xml
#	WooCommerce/src/main/res/layout/fragment_shipping_label_create_custom_package.xml
#	WooCommerce/src/main/res/layout/fragment_shipping_label_create_package.xml
#	WooCommerce/src/main/res/layout/fragment_shipping_packages_selector.xml

AnirudhBhat committed:
Merge branch 'issue/toolbar-elevation' into issue/10649-printing-instructions-dialog

AnirudhBhat committed:
Merge branch 'issue/10649-printing-instructions-dialog' into issue/10571-resize-filters-fragment-dialog

AnirudhBhat committed:
Merge branch 'issue/10571-resize-filters-fragment-dialog' into issue/10576-save-restore-navigation-state

AnirudhBhat committed:
Merge branch 'issue/10576-save-restore-navigation-state' into issue/10773-collect-payment-method

AnirudhBhat committed:
Add toolbar for AddOrderShipmentTrackingFragment and AddOrderTrackingProviderListFragment

AnirudhBhat committed:
Refactor logic to open a specific order or first order

AnirudhBhat committed:
Merge branch 'issue/10576-save-restore-navigation-state' into issue/10773-collect-payment-method

AnirudhBhat committed:
Merge branch 'issue/10773-collect-payment-method' into issue/toolbar-add-shipment-tracking

AnirudhBhat committed:
Show/Hide clear menu options on filter dialog

AnirudhBhat committed:
Highlight and open the first order when filters are applied in Order list screen

AnirudhBhat committed:
Merge branch 'issue/10576-save-restore-navigation-state' into issue/10773-collect-payment-method

AnirudhBhat committed:
Merge branch 'issue/10773-collect-payment-method' into issue/toolbar-add-shipment-tracking

AnirudhBhat committed:
Make the order detail scrollable even from the margins on tablet

AnirudhBhat committed:
Add a small delay before opening the first order when the user logs in on tablet devices

AnirudhBhat committed:
Merge branch 'trunk' into issue/10568-nav-host-fragment

AnirudhBhat commented:
@pachlava The UI test failure in this PR appears to be related to the `constructor() : super(R.id.toolbar)` in `FilterScreen`. This issue likely stems from our refactoring, where we transitioned to using fragment-specific toolbars in place of a shared toolbar.
cc @samiuelson

AnirudhBhat committed:
Add release notes

AnirudhBhat committed:
Hide shared toolbar again which was deleted by mistake

AnirudhBhat committed:
Hide shared toolbar again which was deleted by mistake

AnirudhBhat committed:
Hide shared toolbar again which was deleted by mistake

AnirudhBhat committed:
Merge branch 'issue/toolbar-shipping-package-selector' into issue/toolbar-create-package-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-create-package-fragment' into issue/toolbar-create-custom-package

AnirudhBhat committed:
Hide shared toolbar again which was deleted by mistake

AnirudhBhat committed:
Merge branch 'issue/toolbar-create-custom-package' into issue/toolbar-address-suggestion-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-address-suggestion-fragment' into issue/customs-fragment-shipping

AnirudhBhat committed:
Hide shared toolbar again which was deleted by mistake

AnirudhBhat committed:
Merge branch 'issue/customs-fragment-shipping' into issue/toolbar-carrier-rates-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-carrier-rates-fragment' into issue/toolbar-create-shipping-label

AnirudhBhat committed:
Merge branch 'issue/toolbar-create-shipping-label' into issue/toolbar-edit-shipping-label

AnirudhBhat committed:
Hide shared toolbar again which was deleted by mistake

AnirudhBhat committed:
Merge branch 'issue/toolbar-edit-shipping-label' into issue/toolbar-edit-shipping-label-package

AnirudhBhat committed:
Hide shared toolbar again which was deleted by mistake

AnirudhBhat committed:
Merge branch 'issue/toolbar-edit-shipping-label-package' into issue/toolbar-edit-shipping-payment

AnirudhBhat committed:
Hide shared toolbar again which was deleted by mistake

AnirudhBhat committed:
Merge branch 'issue/toolbar-edit-shipping-payment' into issue/toolbar-select-payment-method

AnirudhBhat committed:
Merge branch 'issue/toolbar-select-payment-method' into issue/toolbar-payments-hub-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-payments-hub-fragment' into issue/toolbar-card-reader-onboarding-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-card-reader-onboarding-fragment' into issue/refresh-orders-list

AnirudhBhat committed:
Merge branch 'issue/refresh-orders-list' into issue/fix-custom-fields-crash

AnirudhBhat committed:
Merge branch 'issue/10568-nav-host-fragment' into issue/10588-open-order-detail

AnirudhBhat committed:
Merge branch 'issue/10588-open-order-detail' into issue/10569-open-first-order

AnirudhBhat committed:
Merge branch 'issue/10569-open-first-order' into issue/10595-highlight-selected-order

AnirudhBhat committed:
Merge branch 'issue/10595-highlight-selected-order' into issue/toolbar-order-list-detail

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailFragment.kt

AnirudhBhat committed:
Fix merge conflicts

AnirudhBhat committed:
Merge branch 'issue/toolbar-order-list-detail' into issue/toolbar-customer-order-note-editing

AnirudhBhat committed:
Merge branch 'issue/toolbar-customer-order-note-editing' into issue/toolbar-base-address-editing-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-base-address-editing-fragment' into issue/toolbar-add-order-note-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-add-order-note-fragment' into issue/toolbar-print-shipping-label-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-print-shipping-label-fragment' into issue/toolbar-label-format-options-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-label-format-options-fragment' into issue/toolbar-print-shipping-label-info-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-print-shipping-label-info-fragment' into issue/toolbar-print-shipping-label-customs-form-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-print-shipping-label-customs-form-fragment' into issue/toolbar-shipping-label-refund-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-shipping-label-refund-fragment' into issue/toolbar-shipping-package-selector

AnirudhBhat committed:
Merge branch 'issue/toolbar-shipping-package-selector' into issue/toolbar-create-package-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-create-package-fragment' into issue/toolbar-create-custom-package

AnirudhBhat committed:
Merge branch 'issue/toolbar-create-custom-package' into issue/toolbar-address-suggestion-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-address-suggestion-fragment' into issue/customs-fragment-shipping

AnirudhBhat committed:
Merge branch 'issue/customs-fragment-shipping' into issue/toolbar-carrier-rates-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-carrier-rates-fragment' into issue/toolbar-create-shipping-label

AnirudhBhat committed:
Merge branch 'issue/toolbar-create-shipping-label' into issue/toolbar-edit-shipping-label

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelAddressFragment.kt
#	WooCommerce/src/main/res/layout/fragment_edit_shipping_label_address.xml

AnirudhBhat committed:
Merge branch 'issue/toolbar-edit-shipping-label' into issue/toolbar-edit-shipping-label-package

AnirudhBhat committed:
Merge branch 'issue/toolbar-edit-shipping-label-package' into issue/toolbar-edit-shipping-payment

AnirudhBhat committed:
Merge branch 'issue/toolbar-edit-shipping-payment' into issue/toolbar-select-payment-method

AnirudhBhat committed:
Merge branch 'issue/toolbar-select-payment-method' into issue/toolbar-payments-hub-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-payments-hub-fragment' into issue/toolbar-card-reader-onboarding-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-card-reader-onboarding-fragment' into issue/refresh-orders-list

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/payment/CardReaderPaymentDialogFragment.kt

AnirudhBhat committed:
Merge branch 'issue/refresh-orders-list' into issue/fix-custom-fields-crash

AnirudhBhat committed:
Add missing methods during merge conflicts

AnirudhBhat committed:
Merge branch 'issue/fix-custom-fields-crash' into issue/toolbar-refund-fragment

AnirudhBhat committed:
Merge branch 'issue/toolbar-refund-fragment' into issue/toolbar-elevation

AnirudhBhat committed:
Merge branch 'issue/toolbar-elevation' into issue/10649-printing-instructions-dialog

AnirudhBhat committed:
Merge branch 'issue/10649-printing-instructions-dialog' into issue/10571-resize-filters-fragment-dialog

AnirudhBhat committed:
Merge branch 'issue/10571-resize-filters-fragment-dialog' into issue/10576-save-restore-navigation-state

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelAddressFragment.kt
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/ShippingLabelAddressSuggestionFragment.kt
#	WooCommerce/src/main/res/layout/fragment_order_list.xml

AnirudhBhat committed:
Merge branch 'issue/10576-save-restore-navigation-state' into issue/10773-collect-payment-method

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt

AnirudhBhat committed:
Merge branch 'issue/10773-collect-payment-method' into issue/toolbar-add-shipment-tracking

AnirudhBhat committed:
Remove unnecessary code

AnirudhBhat committed:
Hide shared toolbar

AnirudhBhat committed:
Add extension function to detect whether the device is tablet of not on Activity

AnirudhBhat committed:
Add analytics constants

AnirudhBhat committed:
Track device type when orders tab is clicked

AnirudhBhat committed:
Track device type when open orders is clicked

AnirudhBhat committed:
Track device type when collect payment is clicked

AnirudhBhat committed:
Change the navigation logic for opening web view from order detail screen

AnirudhBhat committed:
Add toolbar title for edit shipping label address and shipping label address screens which got removed during merge conflicts

AnirudhBhat committed:
Add tests to verify we track correct properties

AnirudhBhat committed:
Merge branch 'issue/toolbar-add-shipment-tracking' into issue/10806-add-analytics

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditViewModel.kt

AnirudhBhat committed:
Merge branch 'trunk' into issue/10568-nav-host-fragment

AnirudhBhat committed:
Merge pull request #10592 from woocommerce/issue/10588-open-order-detail

Business logic to open order detail on order item click in 2 pane layout

AnirudhBhat committed:
Merge pull request #10642 from woocommerce/issue/10569-open-first-order

Select the first order by default in tablet

AnirudhBhat committed:
Add proper color resource for order item highlighted

AnirudhBhat committed:
Merge pull request #10643 from woocommerce/issue/10595-highlight-selected-order

Highlight the selected order in the order list

AnirudhBhat committed:
Make the menu item click listener function private

AnirudhBhat committed:
Merge pull request #10740 from woocommerce/issue/toolbar-order-list-detail

Add toolbar for Order list and Order detail

AnirudhBhat committed:
Merge pull request #10741 from woocommerce/issue/toolbar-customer-order-note-editing

Add toolbar for CustomerOrderNoteEditingFragment

AnirudhBhat committed:
Merge pull request #10742 from woocommerce/issue/toolbar-base-address-editing-fragment

Add toolbar for BaseAddressEditingFragment

AnirudhBhat committed:
Merge pull request #10743 from woocommerce/issue/toolbar-add-order-note-fragment

Add toolbar for AddOrderNoteFragment

AnirudhBhat committed:
Merge pull request #10744 from woocommerce/issue/toolbar-print-shipping-label-fragment

Add toolbar to PrintShippingLabelFragment

AnirudhBhat committed:
Merge pull request #10745 from woocommerce/issue/toolbar-label-format-options-fragment

Add toolbar to label format options fragment

AnirudhBhat committed:
Merge pull request #10746 from woocommerce/issue/toolbar-print-shipping-label-info-fragment

Add toolbar to PrintShippingLabelInfoFragment

AnirudhBhat committed:
Merge pull request #10747 from woocommerce/issue/toolbar-print-shipping-label-customs-form-fragment

Add toolbar to print shipping label customs form fragment

AnirudhBhat committed:
Merge pull request #10748 from woocommerce/issue/toolbar-shipping-label-refund-fragment

Add toolbar to ShippingLabelRefundFragment

AnirudhBhat committed:
Merge pull request #10749 from woocommerce/issue/toolbar-shipping-package-selector

Add toolbar for ShippingPackageSelectorFragment

AnirudhBhat committed:
Merge pull request #10750 from woocommerce/issue/toolbar-create-package-fragment

Add toolbar for ShippingLabelCreatePackageFragment

AnirudhBhat committed:
Merge pull request #10751 from woocommerce/issue/toolbar-create-custom-package

Add toolbar for create custom package screen

AnirudhBhat committed:
Add toolbar title

AnirudhBhat committed:
Remove unused private property

AnirudhBhat committed:
Merge pull request #10752 from woocommerce/issue/toolbar-address-suggestion-fragment

Add toolbar for ShippingLabelAddressSuggestionFragment

AnirudhBhat committed:
Merge pull request #10753 from woocommerce/issue/customs-fragment-shipping

Add toolbar for ShippingCustomsFragment

AnirudhBhat committed:
Merge pull request #10754 from woocommerce/issue/toolbar-carrier-rates-fragment

Add toolbar for ShippingCarrierRatesFragment

AnirudhBhat committed:
Merge pull request #10755 from woocommerce/issue/toolbar-create-shipping-label

Add toolbar for CreateShippingLabelFragment

AnirudhBhat committed:
Merge pull request #10756 from woocommerce/issue/toolbar-edit-shipping-label

Add toolbar for EditShippingLabelAddressFragment

AnirudhBhat committed:
Merge pull request #10757 from woocommerce/issue/toolbar-edit-shipping-label-package

Add toolbar to edit shipping label package screen

AnirudhBhat committed:
Merge pull request #10758 from woocommerce/issue/toolbar-edit-shipping-payment

Add toolbar to edit shipping payment screen

AnirudhBhat committed:
Merge pull request #10765 from woocommerce/issue/toolbar-select-payment-method

Add toolbar for SelectPaymentMethodFragment

AnirudhBhat commented:
@pachlava The UI test failure in this PR appears to be related to the `constructor() : super(R.id.toolbar)` in `FilterScreen`. This issue likely stems from our recent refactoring, where we transitioned to using fragment-specific toolbars in place of a shared toolbar.

AnirudhBhat committed:
Merge pull request #10766 from woocommerce/issue/toolbar-payments-hub-fragment

Add toolbar for PaymentsHubFragment screen

AnirudhBhat committed:
Merge pull request #10767 from woocommerce/issue/toolbar-card-reader-onboarding-fragment

Add toolbar for CardReaderOnboardingFragment

AnirudhBhat committed:
Add methods that got removed by mistake

AnirudhBhat committed:
Merge pull request #10768 from woocommerce/issue/refresh-orders-list

Refresh orders list on a successfull IPP transaction

AnirudhBhat committed:
Merge pull request #10769 from woocommerce/issue/fix-custom-fields-crash

Fix a crash on custom fields view

AnirudhBhat committed:
Merge pull request #10770 from woocommerce/issue/toolbar-refund-fragment

Add toolbars to refund screens

AnirudhBhat committed:
Merge pull request #10771 from woocommerce/issue/toolbar-elevation

Add toolbar elevation

AnirudhBhat committed:
Merge pull request #10662 from woocommerce/issue/10649-printing-instructions-dialog

Printing receipts instructions screen as dialog on tablets

AnirudhBhat committed:
Merge pull request #10663 from woocommerce/issue/10571-resize-filters-fragment-dialog

Resize filter fragment into dialog in tablet

AnirudhBhat committed:
Remove duplicate properties

AnirudhBhat committed:
Merge pull request #10690 from woocommerce/issue/10576-save-restore-navigation-state

Save Order list/Order detail navigation state when the fragment recreates

AnirudhBhat committed:
Remove duplicate properties

AnirudhBhat committed:
Navigate back to the order list screen on tablets after payment has been completed

AnirudhBhat committed:
Merge pull request #10774 from woocommerce/issue/10773-collect-payment-method

Collect payment button while creating a new order navigates to the Order list instead of Select payment method

AnirudhBhat committed:
Merge pull request #10783 from woocommerce/issue/toolbar-add-shipment-tracking

Add toolbar for AddOrderShipmentTrackingFragment and AddOrderTrackingProviderListFragment

AnirudhBhat committed:
Merge pull request #10812 from woocommerce/issue/10806-add-analytics

Add analytics to track device type on some screens

AnirudhBhat committed:
Remove unused import

AnirudhBhat committed:
Make the test pass

AnirudhBhat committed:
Fix failing tests

AnirudhBhat committed:
Add documentation that explains how to optimise screens for tablets

samiuelson commented on RELEASE-NOTES.txt:
```diff
@@ -2,6 +2,11 @@
 -----
+- [***] The Order Listing and Order Detail screens have been optimized for tablet use, showcasing a two-pane layout on such devices. [https://github.com/woocommerce/woocommerce-android/pull/10591]
+
+17.3
+-----
+
```
```suggestion
```
💡 Duplicate

AnirudhBhat replied:
Correct. Fixed here: https://github.com/woocommerce/woocommerce-android/pull/10591/commits/5aab6a91ce70aebafa8697469923774268db1ba7

samiuelson commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt:
```diff
@@ -899,10 +929,34 @@ class MainActivity :
     }
 
-    private fun navigateToWebView(event: ViewUrlInWebView) {
+    fun navigateToWebView(event: ViewUrlInWebView) {
```
```suggestion
    private fun navigateToWebView(event: ViewUrlInWebView) {
```
💡 This should be private

AnirudhBhat replied:
Good catch: Done here: https://github.com/woocommerce/woocommerce-android/pull/10591/commits/b84663b290bba80a37f959f9ef0fb92c855f23cf

samiuelson commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivityViewModel.kt:
```diff
@@ -398,7 +398,11 @@ class MainActivityViewModel @Inject constructor(
+    data class ViewUrlInWebView(
+        val url: String,
+        val urlsToTriggerExit: Array<String>? = null,
+        val title: String? = null,
```
```suggestion
```
❓ These props are not used anywhere

AnirudhBhat replied:
Correct. Removed it here: https://github.com/woocommerce/woocommerce-android/pull/10591/commits/967ad7e9a3678e8075d5ee7b1b599f712402bb4d

AnirudhBhat committed:
Remove duplicate release versions

AnirudhBhat committed:
Remove unnecessary arguments

AnirudhBhat committed:
Remove unnecessary arguments

samiuelson commented:
@AnirudhBhat, as shared in the test charter with more details (pdfdoF-4C0-p2#comment-5822), there are the following navigation scenarios that I believe should be fixed before release:

##### Issue 1
1. Open the app, go to Orders, and wait for the first order to load on the right pane
2. Click back – you'll be redirected to My store
3. Go to Orders again
Observe order is not loading – the pane on the right side displays shimmer animation endlessly.

##### Issue 2
1. Open the app in split mode, go to Orders, and wait for the first order to load on the right pane
2. Adjust Woo app's window size, so that it displays single pane only, the order details screen shuold become visible
3. Click arrow back on the toolbar
Observe there is enless shimmer animation displayed on the screen instead of the orders list screen. We are not able to get back to orders list nor by clicking back arrow navigation or system back press.

##### Issue 3
1. Open the app, go to Orders, and wait for the first order to load on the right pane
2. Click search icon and enter search phrase, e.g. existing order number
3. Click on the search result to load it in the right-side pane
Observe that the order list shows only search results (not the full list of orders) and there is no way to go back to the full orders list. Clicking "back" redirects to My Store instead of going back to full list.

AnirudhBhat committed:
Handle back press and clear selected order id on tablet devices. This is done so that the endless shimmer on the order detail screen bug is resolved.

AnirudhBhat committed:
Do not clear the search bar when the searched order item is clicked on tablets. We need the search bar to go back in tablets

AnirudhBhat commented:
@samiuelson Thanks for the review and testing! Really appreciate it!

I have fixed issue 1 and 3 here: https://github.com/woocommerce/woocommerce-android/pull/10591/commits/f94ef0aaa2502f7b6094bb15cac310a5228d1e50 https://github.com/woocommerce/woocommerce-android/pull/10591/commits/7d16c9903a127cf3e580c9401fb54164d7a968d1

I need additional time to resolve issue 2. Given that it's an edge case, occurring only during the transition from tablet to non-tablet mode, I suggest proceeding with the release. The impact on users is minimal, as navigating to a different tab and returning to the orders tab serves as a workaround. What are your thoughts?

samiuelson reviewed (approved):
Good work, @AnirudhBhat! LGTM. 

It looks like the UI tests (e2eRealApiOrdersFilter) should still be updated to make the checks pass.

Please also make sure to add any pending issues to the project board.

AnirudhBhat committed:
Merge branch 'trunk' into issue/10568-nav-host-fragment

AnirudhBhat committed:
Remove the filters related logic in the test teardown method as it is failing due to the recent toolbar refactoring work.

AnirudhBhat committed:
Ignore the e2eRealordersapi test

AnirudhBhat committed:
Make changes related to how search view behaves on tablet and on phones

AnirudhBhat committed:
Handle back press when the search view is open

AnirudhBhat committed:
Remove code that is resulting in test failure. This will be investigated later.

AnirudhBhat committed:
Ignore failing test. It will be investigated further.

AnirudhBhat committed:
Remove shared toolbar and add fragment specific for customer note screen in order creation screen.

AnirudhBhat committed:
Ignoring another test which needs fixing after our recent toolbar refactoring

samiuelson commented:
Adding the "do not merge" label until we make a final decision on whether to include it in this week's release or not. cc @malinajirka

AnirudhBhat commented:
@pachlava @samiuelson Heads up. I have added `@Ignore` annotation for couple of UI tests that need some fix due to our recent toolbar refactoring. https://github.com/woocommerce/woocommerce-android/pull/10591/commits/b85d5625f592574fadb03273f2e5feb5ba9f510a

malinajirka commented:
Thank you both! We have found some issues today, which we fixed. However, it seems having an additional week for testing will increase our confidence. Moreover, I don't think we should merge a PR with broken tests - that beats their purpose. Let's move the target date to the next week.

kidinov committed:
Toolbar above ScrollChildSwipeRefreshLayout

kidinov committed:
Revert "Toolbar above ScrollChildSwipeRefreshLayout"

This reverts commit 81b246cc9133a32d439ed4a2823efe1fd7cf258a.

kidinov commented on WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/OrdersUITest.kt:
```diff
@@ -48,6 +49,7 @@ class OrdersUITest : TestBase() {
     }
 
     @Test
+    @Ignore
```
Just noticed that 2 tests are ignored now

samiuelson committed:
Open order details in a single-top

When in two-pane mode, we want to avoid stacking multiple OrderDetailFragments in the right pane's NavHostFragment

samiuelson committed:
Fix handling navigating up

samiuelson committed:
Clean up code

samiuelson committed:
Reorder imports

samiuelson committed:
Fix empty state displaying

samiuelson committed:
Reorder imports

samiuelson committed:
Fix switching between empty state and results

samiuelson committed:
Merge branch 'trunk' into issue/10568-nav-host-fragment

samiuelson committed:
Merge branch 'trunk' into issue/10568-nav-host-fragment

samiuelson committed:
Merge branch 'issue/10568-nav-host-fragment' into bug/10861-dynamic-screen-size-change

malinajirka committed:
Update FluxC Version to refresh order list automatically

malinajirka committed:
Revert manual requests to refresh order list

samiuelson committed:
Merge branch 'issue/10568-nav-host-fragment' into bug/10861-dynamic-screen-size-change

samiuelson committed:
Fix navigation up from OrderDetailFragment

Commit:
Fix order list reload on search view expand

Ensure orders are reloaded only when the search query is cleared, not when the search view is expanded. This prevents unnecessary reloads and improves user experience.

Commit:
Added toolbar to the refund details layout.

Commit:
Setup the toolbar for the refund details fragment.

Commit:
Adjusted the margins for small tablets in landscape mode.

Commit:
Ensure that only small tablets trigger portrait logic.

malinajirka committed:
Remove duplicate re-fetch of order list

malinajirka committed:
Don't open first item on search

malinajirka committed:
Update FluxC version

malinajirka committed:
Fix detekt

samiuelson committed:
Rename layout ids

malinajirka committed:
Update FluxC hash

malinajirka committed:
Update FluxC hash

samiuelson committed:
Merge pull request #10877 from woocommerce/bug/10862-crash-when-no-orders

[Tablet Support M2] Bug/10862 crash when no orders

samiuelson committed:
Merge branch `issue/10568-nav-host-fragment`

samiuelson committed:
Merge pull request #10886 from woocommerce/issue-10880/fix-refund-products-fragment-toolbar

[Tablet Support M2] Issue 10880 - Add Refund Products Toolbar

samiuelson committed:
Merge `issue/10568-nav-host-fragment`

samiuelson committed:
Update fluxCVersion

samiuelson committed:
Merge branch 'issue/10568-nav-host-fragment' into issue/7166-update-order-list-automatically

samiuelson committed:
Merge branch 'trunk' into issue/10568-nav-host-fragment

samiuelson committed:
Merge pull request #10887 from woocommerce/issue-10860/small-tablets-order-optimization

[Tablet Support M2] Issue 10860 - Improve Order List & Details UI in Small Tablets

samiuelson committed:
Merge branch `issue/10568-nav-host-fragment`

samiuelson committed:
Merge pull request #10888 from woocommerce/issue/7166-update-order-list-automatically

Issue/7166 update order list automatically

malinajirka committed:
Merge pull request #10883 from woocommerce/issue-10863/fix-search-clear-reload

[Tablet Support M2] Issue 10863 - Fix order list reload on search view expand

malinajirka committed:
Merge pull request #10872 from woocommerce/bug/10861-dynamic-screen-size-change

[Tablet Support M2] Bug/10861 Navigation during dynamic screen size changes

Commit:
Fix Concurrency issue by using CopyOnWriteArrayList for observers

Replaces the observers ArrayList with CopyOnWriteArrayList in ProductImageMap to prevent ConcurrentModificationException during concurrent access. This change ensures thread-safe modifications and iterations over the observer list, improving the stability of the product image fetching mechanism.

malinajirka committed:
Merge pull request #10902 from woocommerce/issue/concurrent-modification-exception

[Tablet Support M2] Prevent ConcurrentModificationException in ProductImageMap with CopyOnWriteArrayList

samiuelson committed:
Fix back button handling in details pane

malinajirka committed:
Merge pull request #10907 from woocommerce/bug/navigation-on-back-press-doesnt-pop-details-backstack

[Tablet Support M2] [Bug] Navigation on back press doesn't pop details backstack
